AGENT - GSWTA (#32705)
DataDog/documentation: content/en/getting_started/agent/_index.md

      text: 'Why should I install the Datadog Agent on my cloud instances?'
---

## Overview

This guide introduces the Datadog Agent and covers:

  - [Introduction to the Agent](#the-datadog-agent)
  - [Installing the Agent](#installing-the-agent)
  - [Data collected by the Agent](#data-collected-by-the-agent)
  - [Advanced configurations and features](#advanced-configurations-and-features)
  - [Troubleshooting the Agent](#troubleshooting)

> ## What is the Datadog Agent

Reviewer: Feels weird not to end with a question mark:
```suggestion
## What is the Datadog Agent?
```

The Datadog Agent is a software that runs on your hosts. It collects events and metrics from hosts and sends them to Datadog, where you can analyze your monitoring and performance data. 

The Agent can run on:
- Your local hosts (Windows, MacOS), 
- A containerized environments (Docker, Kubernetes),

> - In on-premises data centers. 

Reviewer: ```suggestion
- Local hosts (Windows, macOS)
- Containerized environments (Docker, Kubernetes)
- On-premises data centers
```

You can also install and configure the Agent using configuration management tools like Chef, Puppet, or Ansible.

> The Agent is able to collect 75 to 100 system-level metrics every 15 to 20 seconds. With additional configuration, the Agent can send live data, logs, and traces from running processes to the Datadog Platform. The Datadog Agent is open source and its source code is available on GitHub at [DataDog/datadog-agent][1].

Reviewer: I know you didn't work on this part, but there's a few things worth changing here for editorial style:
```suggestion
The Agent can collect 75-100 system-level metrics every 15-20 seconds. With additional configuration, it can send live data, logs, and traces from running processes to Datadog. The Datadog Agent is open source, and its source code is available on GitHub at [DataDog/datadog-agent][1].
```

### The Agent configuration file

The Agent's main configuration file is `datadog.yaml`. The required parameters are:

> - Your [Datadog API key][16], which is used to associate your Agent's data with your organization. 

Reviewer: ```suggestion
- Your [Datadog API key][16], which is used to associate the Agent's data with your organization. 
```

- Your [Datadog site][41] ({{< region-param key="dd_site" code="true" >}}).

See the [sample `config_template.yaml` file][23] for all available configuration options. You can adjust the Agent configuration files to take advantage of other Datadog features including tags.

## How to install the Agent

### Prerequisites
1. Create a [Datadog account][15].

2. Have your [Datadog API key][16] on hand.

### Setup

Use [Fleet Automation][39], the Datadog in-app workflow, to install, upgrade, configure, and troubleshoot the Datadog Agent at scale. 

> See the [Agent documentation][40] for how to get started with the Agent for your specific platform.

Reviewer: Consider swapping the order here. Someone just getting started probably won't start with deploying a whole fleet of Agents at scale; they'll probably first install it on one machine to try it out.
```suggestion
See the [Agent documentation][40] for how to get started with the Agent for your specific platform.

Use [Fleet Automation][39], the Datadog in-app workflow, to install, upgrade, configure, and troubleshoot the Datadog Agent at scale. 
```

Author: For this one, your comment makes complete sense. They can, however, use Fleet automation (FA) to install on just one Agent. It's that FA also helps with scaling Agent configs. As the PM's prefer we push folks towards FA, I can better clarify this. Let me know what you think!

Reviewer: > For this one, your comment makes complete sense. They can, however, use Fleet automation (FA) to install on just one Agent. It's that FA also helps with scaling Agent configs. As the PM's prefer we push folks towards FA, I can better clarify this. Let me know what you think!

Gotcha, thanks for extra context! Keeping the current order to highlight the higher priority method with FA sounds good!

      text: 'Why should I install the Datadog Agent on my cloud instances?'
---

## Overview

This guide introduces the Datadog Agent and covers:

  - [Introduction to the Agent](#the-datadog-agent)
  - [Installation](#installing-the-agent)
  - [Data collected by the Agent](#data-collected-by-the-agent)
  - [Advanced configurations and features](#advanced-configurations-and-features)
  - [Troubleshooting](#troubleshooting)

## What is the Datadog Agent?

The Datadog Agent is software that runs on your hosts. It collects events and metrics from hosts and sends them to Datadog, where you can analyze your monitoring and performance data. 

The Agent can run on:
- Local hosts (Windows, MacOS), 
- Containerized environments (Docker, Kubernetes),

> - On-premises data centers. 

Reviewer: Avoid separating bullet lists with [commas](https://datadoghq.atlassian.net/wiki/spaces/WRITING/pages/5341511907/Lists?atlOrigin=eyJpIjoiMGU5MzhjYmQ3MGUxNGViOTkxYjliNzBkNGM2ZTA3Y2EiLCJwIjoiYyJ9#:~:text=Do%20not%20use%20semicolons%2C%20commas%2C%20or%20conjunctions%20to%20separate%20the%20bulleted%20or%20numbered%20list%20items.). Lists that read like one continuous sentence can be harder to scan.

Also note the capitalization of macOS:
```suggestion
The Agent can run on:
- Local hosts (Windows, macOS)
- Containerized environments (Docker, Kubernetes)
- On-premises data centers
```

You can also install and configure the Agent using configuration management tools like Chef, Puppet, or Ansible.

> The Agent can collect 75-100 system-level metrics every 15-20 seconds. With additional configuration, it can send live data, logs, and traces from running processes to the Datadog Platform. The Datadog Agent is open source, and its source code is available on GitHub at [DataDog/datadog-agent][1].

Reviewer: ```suggestion
The Agent can collect 75-100 system-level metrics every 15-20 seconds. With additional configuration, it can send live data, logs, and traces from running processes to Datadog. The Datadog Agent is open source, and its source code is available on GitHub at [DataDog/datadog-agent][1].
```

To collect metrics from other technologies, see the [Integrations][9] page.

### Service checks

The Agent is set up to provide the following service checks:

  - `datadog.agent.up`: Returns **OK** if the Agent connects to Datadog.

>   - `datadog.agent.check_status`: Returns **CRITICAL** if an Agent check is unable to send metrics to Datadog, otherwise returns **OK**.

Reviewer: ```suggestion
  - `datadog.agent.check_status`: Returns **CRITICAL** if an Agent check is unable to send metrics to Datadog; otherwise returns **OK**.
```

6. Go to the [Events Explorer page][20] and find the custom tags displayed with the latest Agent Event.

{{% /collapse-content %}} 

{{% collapse-content title="Finding metrics in the Datadog UI" level="h4" expanded=false id="id-for-anchoring" %}}

In the Datadog UI, go to the [Metrics Summary page][22] and search for the metric `datadog.agent.started` or the metric `datadog.agent.running`. If these metrics are not visible right away, it may take a few minutes for the Agent to send the data to the Datadog Platform.

Click on either of the metrics and a Metric panel opens up. This panel shows additional metadata about where these metrics are collected from and any associated tags. Because so far in this walkthrough no tags are configured on this host, you should see only the default tags that Datadog assigns to the metrics including `version` and `host`. See the following section on Agent Configuration Files to learn more about how to add tags.

Explore other default metrics such as `ntp.offset` or `system.cpu.idle`.
{{% /collapse-content %}} 

{{% collapse-content title="Agent overhead" level="h4" expanded=false id="id-for-anchoring" %}}

> The amount of space and resources the Agent takes up depends on the configuration and what data the Agent is configured to send. At the onset, you can expect around 0.08% CPU used on average with a disk space of roughly 880MB to 1.3GB.

Reviewer: ```suggestion
The amount of space and resources the Agent takes up depends on the configuration and what data the Agent is sending. At the onset, you can expect around 0.08% CPU used on average with a disk space of roughly 880MB to 1.3GB.
```
